pandas/tests/arithmetic/conftest.py

```diff
@@ -150,42 +163,86 @@ def box(request):
     """
     return request.param
 
 
-@pytest.fixture(params=[(pd.Index, False),
-                        (pd.Series, False),
+@pytest.fixture(params=[pd.Index, pd.Series, pd.DataFrame, tm.to_array],
+                ids=id_func)
+def box4(request):
+    """
+    Fixture to test on each of Index, Series, DataFrame, and pandas Array
+    classes.
+    """
+    return request.param
+
+
+@pytest.fixture(params=[pd.Index,
+                        pd.Series,
                         (pd.DataFrame, False),
-                        pytest.param((pd.DataFrame, True),
-                                     marks=pytest.mark.xfail(strict=True))],
-                ids=lambda x: x[0].__name__ + '-' + str(x[1]))
-def box_transpose_fail(request):
+                        (pd.DataFrame, True),
+                        tm.to_array],
+                ids=id_func)
+def box5(request):
     """
-    Fixture similar to `box` but testing both transpose cases for DataFrame,
-    with the tranpose=True case xfailed.
+    Like `box4`, but the DataFrame case with box transpose=True and
+    transpose=False
     """
     # GH#23620
     return request.param
 
 
-@pytest.fixture(params=[pd.Index, pd.Series, pd.DataFrame, PeriodArray],
-                ids=lambda x: x.__name__)
-def box_with_period(request):
+@pytest.fixture(params=[pd.Index,
+                        pd.Series,
+                        (pd.DataFrame, False),
+                        pytest.param((pd.DataFrame, True),
+                                     marks=pytest.mark.xfail(strict=True)),
+                        tm.to_array],
+                ids=id_func)
+def box5_tfail(request):
     """
-    Like `box`, but specific to PeriodDtype for also testing PeriodArray
+    Like `box5`, but xfailing the transposed DataFrame case
     """
+    # GH#23620
     return request.param
 
 
-@pytest.fixture(params=[pd.Index, pd.Series, pd.DataFrame, DatetimeArray],
-                ids=lambda x: x.__name__)
-def box_with_datetime(request):
+# Construct tuples of the form (box, tz) over all our box classes and
+#  the timezones in tz_naive_fixture.  We then xfail the cases where the box
+#  is a DataFrame-with-transpose and the timezone is not tz-naive (i.e. None)
+boxes = [pd.Index,
+         pd.Series,
+         (pd.DataFrame, False),
+         (pd.DataFrame, True),
+         tm.to_array]
+
+# copied from pandas.conftest tz_naive_fixture
+TIMEZONES = [None, 'UTC', 'US/Eastern', 'Asia/Tokyo', 'dateutil/US/Pacific',
+             'dateutil/Asia/Singapore']
+
+params = [(x, y) for x in boxes for y in TIMEZONES]
+for n in range(len(params)):
+    tup = params[n]
+    if isinstance(tup[0], tuple) and tup[0][1] is True and tup[1] is not None:
+        # i.e. (DataFrame, True, tzinfo) excluding the no-tzinfo case
+        param = pytest.param(tup, marks=pytest.mark.xfail(strict=True))
+        params[n] = param
+
+
+@pytest.fixture(params=params)
+def box5_and_tz(request):
```

Comment:
Do we think this is really needed? (all the timezones) Because with all combinations, you get a quite combinatorial explosion (using this makes a single test into 30 tests) 
And if you then further combine it with other fixtures ..

(I don't know how much it is used)

Reply:
combinatorial explosion is an entirely reasonable concern.

Note that this fixture is taking the place of existing usages of `box[...], tz_naive_fixture`, so the net effect on the number of cases is not as big as it might look at first glance.  The big benefit is that tests that previously were either skipped or marked with a `# FIXME` are now explicitly+strictly xfailed.  (some of these can be un-xfailed and simplified if/when the issue behind #23730 is fixed)

More generally the timezone fixtures may be leading to more tests than we really need/want, especially given the overhead of pytest collection.  Two thoughts on this I've been mulling over:

- We've gotten rid of a lot of for-loops and replaced them with parametrization/fixtures.  Swinging back in the other direction may be worth considerin.  Using the `--showlocals` flag will cause the pytest error messages to tell us where in the loop an error occurred, obviating one of the disadvantages of looping.

- Instead of using a fixed set of timezone strings, have a smaller randomized set.  We could randomize over the whole space of _possible_ dateutil+pytz+stdlib tzinfos, giving corner cases fewer shadows in which to hide.

Reply:
Yes, in general those fixtures are certainly good.

There is for example one place where it is combined with the `two_hours` fixture (6 different ways for timedelta-like), giving 180 tests. 

Maybe more specific for this case: are both 'US/Eastern' and 'Asia/Tokyo' needed? (did we had bugs in the past that were related to that?) 
And the same for the dateutil ones: did we have bugs in the past in the arithmetics related to having a dateutil timezone?

Reply:
No idea about why tz_naive_fixture includes exactly what it does.

If "the tests are unnecessarily thorough" is the biggest problem we have this week, it's a pretty darn good week.

Reply:
so the timezones that are chosen, have been sources of (since fixed) bugs in the past. We have a pretty good representative sample of various edge cases on timezones and i think we have just about the right amount, not too few and not too many, after all its just 2 per timezone library type + UTC + None.